docs/README.md

```diff
@@ -0,0 +1,137 @@
+# letsencrypt
+
+This role issues Let's Encrypt certificates via DNS-01 or HTTP-01 challenge.
+
+**Please note that you have to use a dns-challenge if you want to issue wildcard certificates.**
+
+For distribution of certificates please implement your own playbooks/roles/tasks
+
+## Installation
+
+```bash
+ansible-galaxy collection install t_systems_mms.letsencrypt
+```
+
+## Providers
+
+The role supports multiple providers for http- and dns-challenges.
+Please see the corresponding readme files for specific variables and examples.
+
+Feel free to contribute with more DNS APIs or more HTTP variants :)
+
+### http-challenge
+
+Following providers for a http-challenge are currently supported.
+
+| Provider                         | Description
+|----------------------------------|-------------
+| [s3](http-challenge/s3.md)       | uses a aws s3 bucket for validation/hash files
+| [local](http-challenge/local.md) | uses a local path at your server/machine for validation/hash files
+
+### dns-challenge
+
+Following providers for a dns-challenge are currently supported.
+
+| Provider                                | Description
+|-----------------------------------------|-------------
+| [AutoDNS](dns-challenge/autodns.md)     |
+| [Azure](dns-challenge/azure.md)         |
+| [hetzner](dns-challenge/hetzner.md)     |
+| [openstack](dns-challenge/openstack.md) |
+| [pebble](dns-challenge/pebble.md)       | used for integration tests in github workflow
+
+## General variables
+
+| Variable                            | Required | Default | Description
+|-------------------------------------|----------|---------|------------
+| **domain configuration**
+| certificate_name                    | yes      |         | name of the resulting certificate. Most useful for wildcard certificates to not have files named '*.example.com' on the filesystem
+| zone                                | yes      |         | zone in which the dns records should be created
+| subject_alt_name                    | yes      |         | if you want to use wildcard-certificates use base name again as otherwise DNS txt record creation could fail
```

Comment:
There is no `base name` I could use again. You mean certificate_name?

Reply:
This is a relic from when we had base_name as option. Will update